spark/src/test/scala/org/apache/comet/CometExpressionSuite.scala

```diff
@@ -2806,6 +2806,26 @@ class CometExpressionSuite extends CometTestBase with AdaptiveSparkPlanHelper {
       }
   }
 
+  test("rand expression with random parameters") {
+    val partitionsNumber = Random.nextInt(10) + 1
+    val rowsNumber = Random.nextInt(500)
+    val seed = Random.nextLong()
+    // use this value to have both single-batch and multi-batch partitions
+    val cometBatchSize = math.max(1, math.ceil(rowsNumber.toDouble / partitionsNumber).toInt)
+    withSQLConf("spark.comet.batchSize" -> cometBatchSize.toString) {
+      withParquetDataFrame((0 until rowsNumber).map(Tuple1.apply)) { df =>
+        val dfWithRandParameters = df.repartition(partitionsNumber).withColumn("rnd", rand(seed))
+        checkSparkAnswer(dfWithRandParameters)
+        val dfWithOverflowSeed =
+          df.repartition(partitionsNumber).withColumn("rnd", rand(Long.MaxValue))
+        checkSparkAnswer(dfWithOverflowSeed)
+        val dfWithNullSeed =
+          df.repartition(partitionsNumber).selectExpr("_1", "rand(null) as rnd")
+        checkSparkAnswer(dfWithNullSeed)
```

Comment:
We should check that the query is actually running natively when Comet is enabled:

```suggestion
        checkSparkAnswerAndOperator(dfWithRandParameters)
        val dfWithOverflowSeed =
          df.repartition(partitionsNumber).withColumn("rnd", rand(Long.MaxValue))
        checkSparkAnswerAndOperator(dfWithOverflowSeed)
        val dfWithNullSeed =
          df.repartition(partitionsNumber).selectExpr("_1", "rand(null) as rnd")
        checkSparkAnswerAndOperator(dfWithNullSeed)
```